### DISCUSSION TITLE
Post processing after external oidc login

### DISCUSSION THREAD

**Discussion by Justincale** [oqtane/oqtane.framework#4901] [Q&A] [answered]
Hi,

After someone logs in via an external oidc provider i wish to run some additional post processing which creates some client session data based on the user that just logged in. What would you suggest would be the way to go for this?

**Comment by sbwalker** [comment, marked as answer]
Oqtane has an event system which allows you to create custom logic that runs after events occur:

https://www.oqtane.org/blog/!/53/oqtane-server-events

However it appears that there is not currently an event for external login. This would be useful and can easily be added for the upcoming 6.0.1 release.

**Reply by Justincale** [reply]
That would be awesome, thanks Shaun

**Comment by sbwalker** [comment]
PR #4904 provides the events
